src/contexts/UserRecordContext.tsx

```diff
@@ -3,21 +3,37 @@ import UserData from '../interfaces/User';
 import { getDocSnapshot$ } from '../lib/firestore';
 import { useUserContext } from './UserContext';
 
-const UserRecordContext = React.createContext<UserData | null>(null);
+type UserRecordState = {
+    pending: boolean;
+    userRecord: UserData | null;
+};
+
+const UserRecordContext = React.createContext<UserRecordState>({
+    pending: true,
+    userRecord: null,
+});
 
 export const UserRecordProvider: React.FC = ({ children }) => {
-    const [userRecord, setUserRecord] = React.useState<UserData | null>(null);
+    const [userRecord, setUserRecord] = React.useState<UserRecordState>({
+        pending: true,
+        userRecord: null,
+    });
 
-    const user = useUserContext();
+    const { pending, user } = useUserContext();
 
     React.useEffect(() => {
-        return getDocSnapshot$(`/users/${user?.uid}`, {
-            next: (snapshot) => {
-                if (snapshot.exists()) {
-                    setUserRecord(snapshot.data() as UserData);
-                }
-            },
-        });
+        if (!pending && user) {
+            return getDocSnapshot$(`/users/${user.uid}`, {
+                next: (snapshot) => {
+                    if (snapshot.exists()) {
+                        setUserRecord({
+                            pending: false,
+                            userRecord: snapshot.data() as UserData,
+                        });
+                    }
+                },
+            });
+        }
```

Comment:
So while testing I noticed that clicking on "Log out" does not reflect properly. The "Log out" is only apparent after refreshing the page.

Jorge and I made some changes and noticed that the following seemed to fix it

```    
React.useEffect(() => {
        if (!pending && user) {
            return getDocSnapshot$(`/users/${user.uid}`, {
                next: (snapshot) => {
                    if (snapshot.exists()) {
                        setUserRecord({
                            pending: false,
                            userRecord: snapshot.data() as UserData,
                        });
                    }
                },
            });
        }
        setUserRecord({
            pending: pending,
            userRecord: null,
        });
    }, [user]); 
```

Reply:
Yeah good catch!